sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AnomalyDetection/ChangeThresholdCondition.cs

```diff
@@ -38,28 +38,37 @@ public ChangeThresholdCondition(double changePercentage, int shiftPoint, bool is
         /// When set to N, sets as reference the data point that's N positions before the current
         /// point. Value must be at least 1.
         /// </summary>
-        public int ShiftPoint { get; }
+        public int ShiftPoint { get; set; }
 
         /// <summary>
         /// Compared to the previous point, the current point is an anomaly if the percentage of change is more than this value.
         /// </summary>
-        public double ChangePercentage { get; }
+        public double ChangePercentage { get; set; }
 
         /// <summary>
         /// Sets whether to detect data within the range or outside the range. If you want to detect fluctuations, select false.
         /// If you want to detect flat lines in your data, select true.
         /// </summary>
         [CodeGenMember("WithinRange")]
-        public bool IsWithinRange { get; }
+        public bool IsWithinRange { get; set; }
 
         /// <summary>
         /// A point is considered an anomaly only when the deviation occurs in the specified direction.
         /// </summary>
-        public AnomalyDetectorDirection AnomalyDetectorDirection { get; }
+        public AnomalyDetectorDirection AnomalyDetectorDirection { get; set; }
 
         /// <summary>
         /// The <see cref="Models.SuppressCondition"/> to be applied to every unexpected data point.
         /// </summary>
-        public SuppressCondition SuppressCondition { get; }
+        public SuppressCondition SuppressCondition { get; set; }
+
+        internal ChangeThresholdConditionPatch GetPatchModel() => new ChangeThresholdConditionPatch()
```

Comment:
curious, the reason you are naming this GetPatchModel is that this method is only used when doing an update scenario?

Reply:
You're right in saying that we'll only use it during Update, but we name it like this because all types accepted by the service for Update operations end with the `Patch` suffix, e.g., `DataFeedDetailPatch`, `HookInfoPatch`, etc.